src/main/java/uk/gov/register/store/postgres/PostgresReadDataAccessLayer.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Override
+    public Iterator<Entry> getIndexEntryIterator(String indexName) {
+        return indexQueryDAO.getIterator(indexName);
+    }
+
+    @Override
+    public Iterator<Entry> getIndexEntryIterator(String indexName, int totalEntries1, int totalEntries2) {
+        return indexQueryDAO.getIterator(indexName, totalEntries1, totalEntries2);
+    }
+
+    @Override
+    public <R> R withEntryIterator(Function<EntryIterator, R> callback) {
+        checkpoint();
+        return EntryIterator.withEntryIterator(entryQueryDAO, entryIterator -> callback.apply(entryIterator));
+    }
+
+    @Override
+    public int getTotalEntries() {
+        checkpoint();
+        return entryQueryDAO.getTotalEntries();
+    }
+
+    @Override
+    public Optional<Instant> getLastUpdatedTime() {
+        checkpoint();
+        return entryQueryDAO.getLastUpdatedTime();
+    }
+
+    // Item Store
+
+    @Override
+    public Optional<Item> getItemBySha256(HashValue hash) {
+        checkpoint();
+        return itemQueryDAO.getItemBySHA256(hash.getValue());
+    }
+
+    @Override
+    public Collection<Item> getAllItems() {
+        checkpoint();
+        return itemQueryDAO.getAllItemsNoPagination();
+    }
+
+    @Override
+    public Iterator<Item> getItemIterator() {
+        checkpoint();
+        return itemQueryDAO.getIterator();
+    }
+
+    @Override
+    public Iterator<Item> getItemIterator(int start, int end) {
+        checkpoint();
+        return itemQueryDAO.getIterator(start, end);
+    }
+
+    // Record Index
+
+    @Override
+    public Optional<Record> getRecord(String key) {
+        checkpoint();
+        return recordQueryDAO.findByPrimaryKey(key);
+    }
+
+    @Override
+    public List<Record> getRecords(int limit, int offset) {
+        checkpoint();
+        return recordQueryDAO.getRecords(limit, offset);
+    }
+
+    @Override
+    public List<Record> findMax100RecordsByKeyValue(String key, String value) {
+        checkpoint();
+        return recordQueryDAO.findMax100RecordsByKeyValue(key, value);
+    }
+
+    @Override
+    public Collection<Entry> findAllEntriesOfRecordBy(String key) {
+        checkpoint();
+        return recordQueryDAO.findAllEntriesOfRecordBy(key);
+    }
+
+    @Override
+    public int getTotalRecords() {
+        checkpoint();
+        return recordQueryDAO.getTotalRecords();
+    }
+
+    // Index
+
+    @Override
+    public Optional<Record> getIndexRecord(String key, String indexName) {
+        Optional<Record> record = indexQueryDAO.findRecord(key, indexName);
+        return record.filter(r -> r.getItems().size() != 0);
+    }
+
+    @Override
+    public List<Record> getIndexRecords(int limit, int offset, String indexName) {
+        return indexQueryDAO.findRecords(limit, offset, indexName);
+    }
+
+    @Override
+    public int getTotalIndexRecords(String indexName) {
+        return indexQueryDAO.getTotalRecords(indexName);
+    }
+
+    protected void checkpoint() {
```

Comment:
This is needed by all the methods in this class. I have made it more obvious that it needs to be overridden by making it abstract. I think this will go away in the next iteration or two.